lessons/08-mcp-advanced/README.md

```diff
@@ -0,0 +1,268 @@
+>
+> See [Lesson 1](../01-intro-to-genai/README.md) if you want to catch up with the story from the beginning. 
+
+> [!NOTE] 
+> While we recommend going through the story (it's fun!), [click here](#interact-with-hedy-lamarr) if you'd prefer to jump straight to the technical content.
+
+You once again travelled back to Ada's mansion. This time Ada was meeting you at the main gates.
+
+**Ada Lovelace**: Well, how did it go?
+
+**You**: Good I think, we managed to get this app working, you showing the progress to Ada.
+
+**Ada Lovelace**: She takes the device up, tries it out and fiddles it and to herself murmurs, mm hm, ah I see, yeah, yeah no good. It won't do I'm afraid, she states a bit louder, lacks a certain finesse.
+
+**You**: I was thinking that too, we need to be able to type or speak to it right?
+
+**Ada Lovelace**: Right, I know just the person to help us. In fact, I'll come with you for this one, been too long since we met. *Time Beetle, Hollywood please, 1940, residence of Hedy Lamarr*.
+
+Everything went dark, colors swirling, moments later your eyes start to make out a scene, a man sitting by a piano next to a woman with dark brown and curly hair both speaking in an excited way gesticulating. 
+
+<div>
+  <img src="./assets/hedy-invention.jpeg" alt="Hedy Lamarr" width="600" />
+</div>
+
+The woman turned around to face Ada standing next to you to exclaim "Ada it is you, it's been far too long".
+
+**Ada Lovelace**: Hedwig dearest, working on your latest invention I assume?
+
+**Hedy Lamarr**: Yes, in fact George and I think we're on to something "frequency hopping" I probably shouldn't say anymore, who knows who's listening.
+
+**Ada Lovelace**: Oh you mean him, pointing at me, he's on his own adventure.
+
+**Hedy Lamarr**: I see, what can I do for you?
+
+**Ada Lovelace**: Actually, we need this device to work a bit better, ideas?
+
+**Hedy Lamarr**: She picked up the device, looked at it from different angles, and you say you've already separated the features from the communication part?
+
+**Ada Lovelace**: Yes, yes we did.
+
+**Hedy Lamarr**: Well then, I would just make the communication part a bit smarter. This makes me think of a conversation I had with a naval cadet as I was trying to sell war bonds. Minsky was his name I think. "What does Human intelligence look like in a machine" was the topic. I have a feeling he'll do something great in this field one day. So yes, give it more intelligence.
+
+**Ada Lovelace**: Right, well you heard Ms Lamarr, get to it.
+
+**You**: Time Beetle, how do we do this?
+
+**Time Beetle**: You can improve the client you've created before with a large language model, LLM.
+
+> Hedy Lamarr was an extraordinary figure, known both for her Hollywood stardom and her remarkable contributions to technology. Born in Vienna, Austria, in 1914, she became a famous actress in the 1930s and 1940s, often referred to as the "most beautiful woman in the world".
```

Comment:
Let's shorten this to the following:

Hedy Lamarr was an extraordinary figure, known both for her Hollywood stardom and her remarkable contributions to technology. Born in Vienna, Austria, in 1914, she became a famous actress in the 1930s and 1940s.